python/gps/algorithm/traj_opt/traj_opt_lqr_python.py

```diff
@@ -189,7 +284,10 @@ def backward(self, prev_traj_distr, traj_info, eta, algorithm, m):
         dU = prev_traj_distr.dU
         dX = prev_traj_distr.dX
 
-        traj_distr = prev_traj_distr.nans_like()
+        if self._update_in_bwd_pass:
```

Comment:
This PR adds quite a lot of code to the backward function. Would it make sense to make it more modular?